receipt/src/androidTest/java/com/hyperwallet/android/receipt/ListReceiptsTest.java

```diff
@@ -0,0 +1,231 @@
+package com.hyperwallet.android.receipt;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.hasDescendant;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withParent;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+
+import static java.net.HttpURLConnection.HTTP_NO_CONTENT;
+import static java.net.HttpURLConnection.HTTP_OK;
+
+import static com.hyperwallet.android.util.EspressoUtils.atPosition;
+
+import android.content.Context;
+import android.content.res.Configuration;
+import android.content.res.Resources;
+import android.widget.TextView;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.rule.ActivityTestRule;
+
+import com.hyperwallet.android.Hyperwallet;
+import com.hyperwallet.android.receipt.repository.ReceiptRepositoryFactory;
+import com.hyperwallet.android.receipt.view.ListReceiptActivity;
+import com.hyperwallet.android.rule.HyperwalletExternalResourceManager;
+import com.hyperwallet.android.rule.HyperwalletMockWebServer;
+import com.hyperwallet.android.util.RecyclerViewCountAssertion;
+import com.hyperwallet.android.util.TestAuthenticationProvider;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Locale;
+
+@RunWith(AndroidJUnit4.class)
+public class ListReceiptsTest {
+
+    @ClassRule
+    public static HyperwalletExternalResourceManager sResourceManager = new HyperwalletExternalResourceManager();
+    @Rule
+    public HyperwalletMockWebServer mMockWebServer = new HyperwalletMockWebServer(8080);
+    @Rule
+    public ActivityTestRule<ListReceiptActivity> mActivityTestRule =
+            new ActivityTestRule<>(ListReceiptActivity.class, true, false);
+
+    @Before
+    public void setup() {
+        Hyperwallet.getInstance(new TestAuthenticationProvider());
+
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("authentication_token_response.json")).mock();
+
+        setLocale(Locale.US);
+    }
+
+    @After
+    public void cleanup() {
+        ReceiptRepositoryFactory.clearInstance();
+    }
+
+    @Test
+    public void testListReceipts_userHasMultipleTransactions() {
```

Comment:
I think we need to add a use case where there's a new transaction type which is not mapped in our transaction types and whatever default value should be displayed

Reply:
Added